plaso/parsers/shared/unified_logging/timesync.py

```diff
@@ -0,0 +1,120 @@
+
+  def __init__(self):
+    """Initialises the parser."""
+    super(TimesyncParser, self).__init__()
+    self.records = []
+
+  def ParseAll(self, parser_mediator, file_entry, file_system):
+    """Finds and parses all the timesync files
+
+    Args:
+      parser_mediator (ParserMediator): mediates interactions between parsers
+          and other components, such as storage and dfvfs.
+      file_system (dfvfs.FileSystem): file system.
+      file_entry (dfvfs.FileEntry): file entry.
+    """
+    path_segments = file_system.SplitPath(file_entry.path_spec.location)
+    timesync_location = file_system.JoinPath(path_segments[:-2] + ['timesync'])
+    kwargs = {}
+    if file_entry.path_spec.parent:
+      kwargs['parent'] = file_entry.path_spec.parent
+    kwargs['location'] = timesync_location
+    timesync_file_path_spec = path_spec_factory.Factory.NewPathSpec(
+        file_entry.path_spec.TYPE_INDICATOR, **kwargs)
+
+    location_glob = None
+    if file_entry.TYPE_INDICATOR == 'ZIP':
+      location_glob = os.path.join(timesync_location, '*.timesync')
+
+    find_spec = file_system_searcher.FindSpec(
+          file_entry_types=[definitions.FILE_ENTRY_TYPE_FILE],
+          location_glob=location_glob)
+
+    path_spec_generator = file_system_searcher.FileSystemSearcher(
+        file_system, timesync_file_path_spec).Find(
+          find_specs=[find_spec])
+
+    for path_spec in path_spec_generator:
+      try:
+        timesync_file_entry = path_spec_resolver.Resolver.OpenFileEntry(
+            path_spec)
+
+      except RuntimeError as exception:
+        message = (
+            'Unable to open timesync file: {0:s} with error: '
+            '{1!s}'.format(path_spec, exception))
+        parser_mediator.ProduceExtractionWarning(message)
+        continue
+      try:
+        timesync_file_object = timesync_file_entry.GetFileObject()
+        self.ParseFileObject(parser_mediator, timesync_file_object)
+      except (IOError, errors.ParseError) as exception:
+        message = (
+            'Unable to parse data block file: {0:s} with error: '
+            '{1!s}').format(path_spec, exception)
+        parser_mediator.ProduceExtractionWarning(message)
+        continue
+
+  def ParseFileObject(self, parser_mediator, file_object):
+    """Parses a shared-cache strings (dsc) file-like object.
+
+    Args:
+      parser_mediator (ParserMediator): a parser mediator.
+      file_object (dfvfs.FileIO): a file-like object to parse.
+
+    Raises:
+      ParseError: if the records cannot be parsed.
+    """
+    boot_record_data_map = self._GetDataTypeMap('timesync_boot_record')
+    sync_record_data_map = self._GetDataTypeMap('timesync_sync_record')
+
+    file_size = file_object.get_size()
+    offset = 0
+    current_boot_record = None
+    while offset < file_size:
+      try:
+        boot_record, size = self._ReadStructureFromFileObject(
+            file_object, offset, boot_record_data_map)
+        offset += size
```

Comment:
if for some reason record_size would change, this would fail this parsing loop. Also trying to parse and fail boot_record_data_map first is an expensive parsing approach